packages/api/src/beacon/routes/beacon/pool.ts

```diff
@@ -227,12 +237,53 @@ export function getDefinitions(_config: ChainForkConfig): RouteDefinitions<Endpo
       url: "/eth/v1/beacon/pool/attestations",
       method: "POST",
       req: {
-        writeReqJson: ({signedAttestations}) => ({body: AttestationListType.toJson(signedAttestations)}),
-        parseReqJson: ({body}) => ({signedAttestations: AttestationListType.fromJson(body)}),
-        writeReqSsz: ({signedAttestations}) => ({body: AttestationListType.serialize(signedAttestations)}),
-        parseReqSsz: ({body}) => ({signedAttestations: AttestationListType.deserialize(body)}),
+        writeReqJson: ({signedAttestations}) => {
+          const fork = config.getForkName(signedAttestations[0].data.slot);
+          return {
+            body:
+              ForkSeq[fork] >= ForkSeq.electra
+                ? AttestationListTypeElectra.toJson(signedAttestations as AttestationListElectra)
+                : AttestationListTypePhase0.toJson(signedAttestations as AttestationListPhase0),
+            headers: {[MetaHeader.Version]: fork},
+          };
+        },
+        parseReqJson: ({body, headers}) => {
+          const versionHeader = fromHeaders(headers, MetaHeader.Version, false);
+          const fork =
+            versionHeader !== undefined
+              ? toForkName(versionHeader)
+              : config.getForkName(Number((body as {data: {slot: string}}[])[0]?.data.slot ?? 0));
```

Comment:
This is something we generally should avoid, but we can fix that when we implement the v2 api

Reply:
correct